android/src/main/java/com/expensify/reactnativekeycommand/KeyCommandModule.java

```diff
@@ -54,35 +55,72 @@ public void onKeyDownEvent(int keyCode, KeyEvent keyEvent) {
         }
 
         WritableMap params = getJsEventParams(keyCode, keyEvent, null);
-        
-        Set<Object> command = new HashSet<Object>();
-        command.add(params.getString("input"));
-        command.add(params.getInt("modifierFlags"));
 
-        if (commandsArray.contains(command)) {
+        if (matchesInput(params)) {
             mJSModule.emit("onKeyCommand", params);
         }
     };
 
+    private boolean matchesInput(ReadableMap command) {
+        /**
+         * command object may have following schema:
+         * - {input: 'f', modifierFlags: 123}
+         * - {input: 'f'}
+         * - {input: '123'}
+         */
+        for (ReadableMap registeredCommand : commandsArray) {
+            boolean hasModifier = registeredCommand.hasKey("modifierFlags");
+            boolean matchInput = registeredCommand.getString("input").equals(command.getString("input"));
+            if (!hasModifier && matchInput) {
+                return true;
+            }
+
+            if (hasModifier) {
+                boolean matchModifier = registeredCommand.getInt("modifierFlags") == command.getInt("modifierFlags");
+                if (hasModifier && matchInput && matchModifier) {
+                    return true;
+                }
+            }
```

Comment:
```suggestion
            boolean matchInput = registeredCommand.getString("input").equals(command.getString("input"));
            boolean matchModifier = hasModifier && registeredCommand.getInt("modifierFlags") == command.getInt("modifierFlags");
            if (hasModifier) {
                return matchInput && matchModifier;
            }
            return matchInput;
```

Reply:
You can't do that. `registeredCommand.getInt("modifierFlags")` will throw an error if `modifierFlags` don't exist, therefore we first need to check for that condition.

Reply:
extra `hasModifier` check is removed